app/src/main/java/com/example/tasky/agenda/presentation/agenda_list/AgendaViewModel.kt

```diff
@@ -45,10 +45,15 @@ class AgendaViewModel(
     private val localReminderDataSource: RoomLocalReminderDataSource,
 ) : ViewModel() {
 
+    // Moved initializeData from _state.onStart to init because I want to call that function only one
+    // not every time I navigate back to AgendaListScreen
```

Comment:
Or, you could keep it in onStart and just guard with a simple boolean flag. That way the data only initializes once, but you don’t risk stacking multiple collectors every time the screen rotates or comes back into view:
```kotlin
private var initialized = false

val state = _state
    .onStart {
        if (!initialized) {
            initializeData()
            initialized = true
        }
    }
```